dagshub_annotation_converter/ir/image/annotations/base.py

```diff
@@ -0,0 +1,61 @@
+from abc import abstractmethod
+from typing import Optional
+
+from pydantic import BaseModel
+from typing_extensions import Self
+
+from dagshub_annotation_converter.ir.image import NormalizationState
+
+
+class IRAnnotationBase(BaseModel):
+    """
+    Common class for all intermediary annotations
+    """
+
+    filename: Optional[str] = None
+
+    category: str
+    image_width: int
+    image_height: int
+    state: NormalizationState
+    imported_id: Optional[str] = None
+
+    def with_filename(self, filename: str) -> Self:
+        self.filename = filename
+        return self
+
+    def normalized(self) -> Self:
+        """
+        Returns a copy with all parameters in the annotation normalized
+        """
+        if self.state == NormalizationState.NORMALIZED:
+            return self.model_copy()
+        normalized = self.model_copy()
+        normalized._normalize()
+        normalized.state = NormalizationState.NORMALIZED
+        return normalized
+
+    @abstractmethod
+    def _normalize(self):
+        """
+        Every annotation should implement this to normalize itself
+        """
+        ...
+
+    def denormalized(self) -> Self:
+        """
+        Returns a copy with all parameters in the annotation denormalized
+        """
+        if self.state == NormalizationState.DENORMALIZED:
+            return self.model_copy()
+        denormalized = self.model_copy()
+        denormalized._denormalize()
+        denormalized.state = NormalizationState.DENORMALIZED
+        return denormalized
```

Comment:
Normalized/denormalized are pretty common terms in annotation space, see docs for pybboxes: https://github.com/devrimcavusoglu/pybboxes?tab=readme-ov-file#conversion
Also YOLO paper: https://arxiv.org/pdf/1506.02640 (ctrl+f "normalize")

Maybe adding comments + renaming it from `State` to something better... I guess since there's no way we're extending the binary normalized/denormalized, I can swap the Enum for `is_normalized` boolean and that'll be enough.

I honestly can't come up with a satisfying name for this

Reply:
I guess renaming it from State to CoordinateStyle is enough. And I would leave the enum rather than a boolean to future proof ourselves with any bizarre scheme that might come in the future.